static/js/containers/DashboardPage.js

```diff
@@ -229,6 +237,36 @@ class DashboardPage extends React.Component {
     }
   };
 
+  handleCoupon = () => {
+    const { coupons, dispatch, location: { query } } = this.props;
+
+    if (!query.coupon) {
+      // If there's no coupon code in the URL query parameters,
+      // there's nothing to do.
+      return;
+    }
+
+    if ( coupons.fetchPostStatus !== undefined ) {
+      // If we've already launched a coupon fetch, don't launch another one
+      return;
+    }
+
```

Comment:
OK, I wasn't aware you can't abort fetches -- good to know. Rather than ensuring that `fetchGetStatus === FETCH_SUCCESS`, wouldn't it be better to ensure that `fetchGetStatus !== FETCH_PROCESSING`? The fetch can fail, and we don't want that failure to prevent future fetches.